stbt_auto_selftest.py

```diff
@@ -4,12 +4,92 @@
 """
 
 import argparse
+import codecs
+import re
+import StringIO
 import sys
+import traceback
 
 
 def main(argv):
     parser = argparse.ArgumentParser()
     parser.parse_args(argv[1:])
 
+
+def update_doctests(infilename, outfile):
+    """
+    Updates a file with doctests in it but no results to have "correct" results.
+    """
+    module = import_by_filename(infilename)
+    if isinstance(outfile, str):
+        outfile = open(outfile, 'w')
+    with codecs.open(infilename, 'r', 'utf-8') as infile:
+        for line in infile:
+            m = re.match(r'\s*>>> (.*)\n', line)
+            if m:
+                cmd = m.group(1)
+            else:
+                outfile.write(("%s" % line).encode('utf-8'))
+                continue
+
+            if line.endswith(' # remove-if-false\n'):
+                line = line.replace(' # remove-if-false\n', '\n')
+                remove_if_false = True
+            else:
+                remove_if_false = False
+
+            oldstdout = sys.stdout
+            io = StringIO.StringIO()
+            result = None
+            try:
+                sys.stdout = io
+                try:
+                    result = eval(cmd, module.__dict__)  # pylint: disable=W0123
+                except SyntaxError:
+                    exec cmd in module.__dict__  # pylint: disable=W0122
```

Comment:
Might want to warn or fail if somebody used `# remove-if-false` with an expression?

(If the `exec` path is taken here, `result` will be `None`. So if you used `# remove-if-false` anything printed by the selftest-expression will be removed from the outfile.)